lib/sgtn-client/loader/source_comparer.rb

```diff
@@ -2,17 +2,22 @@
 # SPDX-License-Identifier: EPL-2.0
 
 module SgtnClient
-  autoload :Source, 'sgtn-client/loader/local_source_bundle'
+  autoload :StringUtil, 'sgtn-client/util/string-util'
 end
+
 module SgtnClient::TranslationLoader::SourceComparer
   def load_bundle(component, locale)
-    # source locale always uses source bundles
-    return SgtnClient::Source.load_bundle(component) if SgtnClient::LocaleUtil.is_source_locale(locale)
+    # source locale and old source locale don't need comparison because they are bases of comparison
+    if SgtnClient::LocaleUtil.cache_to_real_map.key?(locale)
+      return super(component, SgtnClient::LocaleUtil.cache_to_real_map[locale])
+    end
+
+    old_source_bundle_thread = Thread.new { load_bundle(component, SgtnClient::LocaleUtil::OLD_SOURCE_LOCALE) }
+    source_bundle_thread = Thread.new { load_bundle(component, SgtnClient::LocaleUtil.get_source_locale) }
```

Comment:
No need to fetch multiple times since source is in cache and does not expire.
```suggestion
```

Reply:
fixed